heartbeat/monitors/monitor_test.go

```diff
@@ -100,19 +100,22 @@ func TestDuplicateMonitorIDs(t *testing.T) {
 	// Would fail if the previous newMonitor didn't free the monitor.id
 	m1, m1Err := makeTestMon()
 	require.NoError(t, m1Err)
-	_, m2Err := makeTestMon()
-	require.Error(t, m2Err)
+	m2, m2Err := makeTestMon()
+	// Change the name so we can ensure that this is the currently active monitor
+	m2.stdFields.Name = "MON2!!!"
+	// This used to trigger an error, but shouldn't any longer, we just log
+	// the error, and ensure the last monitor wins
+	require.NoError(t, m2Err)
```

Comment:
can we also check if previous monitor is stopped?

Reply:
We do check for this at the end of the test by checking the number of stops invoked